client/src/components/AddScheduleForm.tsx

```diff
@@ -0,0 +1,204 @@
+import { Button } from '@/components/ui/button';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import { Checkbox } from './ui/checkbox';
+import React, { useEffect, useState } from 'react';
+import { Medication } from 'data';
+import { Progress } from '@/components/ui/progress';
+
+const days = [
+  {
+    id: 'Monday',
+    label: 'Monday',
+  },
+  {
+    id: 'Tuesday',
+    label: 'Tuesday',
+  },
+  {
+    id: 'Wednesday',
+    label: 'Wednesday',
+  },
+  {
+    id: 'Thursday',
+    label: 'Thursday',
+  },
+  {
+    id: 'Friday',
+    label: 'Friday',
+  },
+  {
+    id: 'Saturday',
+    label: 'Saturday',
+  },
+  {
+    id: 'Sunday',
+    label: 'Sunday',
+  },
+];
+
+type Props = {
+  medication: Medication;
+  onScheduleComplete: (medication: Medication) => void;
+};
+
+export function AddScheduleForm({ medication, onScheduleComplete }: Props) {
+  const [checkedState, setCheckedState] = useState<boolean[]>(
+    new Array(days.length).fill(false)
+  );
+  const [timesPerDay, setTimesPerDay] = useState<string>('');
+  const [error, setError] = useState<unknown>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [progress, setProgress] = useState(0);
+
+  // useEffect(() => {
+  //   setUnscheduleMeds(
+  //     medications.filter((medication) => !medication.scheduled)
+  //   );
+  // }, [medications]);
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout;
+    if (isLoading) {
+      timer = setInterval(() => {
+        setProgress((prevProgress) => {
+          if (prevProgress >= 100) {
+            clearInterval(timer);
+            return 100;
+          }
+          return prevProgress + 25;
+        });
+      }, 1000);
+    }
+    return () => clearInterval(timer);
+  }, [isLoading]);
+
+  function handleCheckedChange(position: number) {
+    const updatedCheckedState = checkedState.map((item, index) =>
+      index === position ? !item : item
+    );
+    setCheckedState(updatedCheckedState);
+  }
+
+  async function handleSubmit(e: React.FormEvent) {
+    setIsLoading(true);
+    setProgress(0);
+    e.preventDefault();
```

Comment:
Functionally, I don't think it makes a difference, but usually the first thing you ever want to do is `e.preventDefault()` just to make sure as soon as the form gets submitted you prevent the default action right away